pkgs/by-name/si/simplemoji/package.nix

```diff
@@ -0,0 +1,64 @@
+{
+  autoPatchelfHook,
+  fetchFromGitHub,
+  lib,
+  libxkbcommon,
+  fontconfig,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+  wayland,
+  xorg,
+  noto-fonts-color-emoji,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "simplemoji";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "SergioRibera";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-MU31gRikMqXelWjwjMPZtfbDpybYlPTFr1u9w2843Kc=";
+  };
+
+  cargoHash = "sha256-NvNNc+V7wloPxM532uNrXoou4vnKLhFK29L2/17eB98=";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    pkg-config
+  ];
+
+  buildInputs = [
+    stdenv.cc.cc.libgcc or null # libgcc_s.so
+  ] ++ lib.optionals stdenv.hostPlatform.isLinux [
+    fontconfig.dev
+    libxkbcommon.dev
+    wayland
+    xorg.libxcb
+    xorg.libXrandr
+    wayland # libwayland-client.so
+    xorg.libX11 # libX11.so
+    xorg.libXcursor # libXcursor.so
+    xorg.libXi # libXi.so
+  ];
+
+  runtimeDependencies = [
+    noto-fonts-color-emoji
+  ] ++ lib.optionals stdenv.hostPlatform.isLinux [
+    wayland # libwayland-client.so
+    libxkbcommon # libxkbcommon.so
+    xorg.libX11 # libX11.so
+    xorg.libXcursor # libXcursor.so
+    xorg.libXi # libXi.so
```

Comment:
```suggestion
    wayland
    libxkbcommon
    xorg.libX1
    xorg.libXcursor
    xorg.libXi
```

Reply:
Looks similar to what is done [here](https://github.com/NixOS/nixpkgs/pull/319461). I don't know if all of those runtime dependencies are necessary, so you might want to check with a tool like `strace`.

Reply:
Yep, I was reviewing some PRs to see how to put together the file and from there I took it, I left the comments because I found them useful.

Originally I took the necessary dependencies from the original package [here](https://github.com/SergioRibera/Simplemoji/blob/main/nix/default.nix)